src/crimson/admin/osd_admin.cc

```diff
@@ -131,4 +132,34 @@ class AssertAlwaysHook : public AdminSocketHook {
 };
 template std::unique_ptr<AdminSocketHook> make_asok_hook<AssertAlwaysHook>();
 
+/**
+* A Seastar admin hook: fetching the values of configured metrics
+*/
+class SeastarMetricsHook : public AdminSocketHook {
+public:
+ SeastarMetricsHook()  :
+   AdminSocketHook("perf dump_seastar",
+      "",
+      "dump current configured seastar metrics and their values")
+ {}
+ seastar::future<tell_result_t> call(const cmdmap_t& cmdmap,
+             std::string_view format,
+             ceph::bufferlist&& input) const final
+ {
+   std::unique_ptr<Formatter> f{Formatter::create(format, "json-pretty", "json-pretty")};
+   f->open_object_section("perf_dump_seastar");
+   for (const auto& mf : seastar::metrics::impl::get_value_map()) {
```

Comment:
why do you access the impl? isn;t the data available via seastar::metrics itself?

Reply:
I couldn't find another way to access both the metric name and the metric value. `get_value_map` returns a data structure containing the mapping between metric families and their values, and it is exposed only via `seastar::metrics::impl` to my knowledge.

Reply:
@AmnonHanuhov how about using
```
#include <seastar/core/scollectd_api.hh>

/// ...
for (const auto& mf : seastar::scollectd::get_value_map()) {
  /// ...
}
```